### PR TITLE
OAK-9989: introduce shaded Guava subproject

### PR DESCRIPTION
**Pull request by reschke**
This is a subset of #877 which just introduces the new module without actually using it yet (to make it easier to integrate into dependant projects).

### CODE REVIEW

File: oak-shaded-guava/src/main/java/org/apache/jackrabbit/oak/guava/package-info.java
Comment (mreutegg) [inline oak-shaded-guava/src/main/java/org/apache/jackrabbit/oak/guava/package-info.java:1]: This must be move to org/apache/jackrabbit/guava and updated accordingly.

Why is this file needed?

**Review by mreutegg — CHANGES_REQUESTED** [review]

File: oak-shaded-guava/src/main/java/org/apache/jackrabbit/oak/guava/package-info.java
Comment (reschke) [reply oak-shaded-guava/src/main/java/org/apache/jackrabbit/oak/guava/package-info.java:1]: good point.

I'm not sure it's needed, but I wanted a place to embed the usage warning. Let me check whether it's actually embedded into the Javadoc.

Comment (reschke) [reply oak-shaded-guava/src/main/java/org/apache/jackrabbit/oak/guava/package-info.java:1]: Ok, it's indeed unused. Removed.

**Review by anchela — APPROVED** [review]

**Review by mreutegg — APPROVED** [review]
